### ISSUE TITLE
Aggregates ignore `IGNORE NULLS` and `ORDER BY` on arguments

### ISSUE THREAD

**Issue by alamb** [apache/arrow-datafusion#9924]
### Describe the bug

DataFusion now supports providing ordering and nulls information to aggregates

For example
```sql
❯ select first_value(column1 ORDER BY column2) FROM (values (1,2), (3,4), (-1,0)) ;
+----------------------+
| FIRST_VALUE(column1) |
+----------------------+
| -1                   |
+----------------------+

❯ select first_value(column1 ORDER BY column2) IGNORE NULLS FROM (values (1,2), (3,4), (null,0)) ;
+----------------------+
| FIRST_VALUE(column1) |
+----------------------+
| 1                    |
+----------------------+
1 row in set. Query took 0.002 seconds.
```

### To Reproduce

```sql
❯ select count(*) from (values (1), (null), (2));
+----------+
| COUNT(*) |
+----------+
| 3        |
+----------+
1 row in set. Query took 0.039 seconds.

❯ select count(*) IGNORE NULLS from (values (1), (null), (2));
+----------+
| COUNT(*) |
+----------+
| 3        |
+----------+
1 row in set. Query took 0.001 seconds.
```

Also, for ordering

```sql
❯ select avg(column1 ORDER BY column2) FROM (values (1,2), (3,4), (null,0)) ;
+--------------+
| AVG(column1) |
+--------------+
| 2.0          |
+--------------+
1 row in set. Query took 0.008 seconds.
```

### Expected behavior

I expect

```sql
❯ select count(*) IGNORE NULLS from (values (1), (null), (2));
```

To error with "IGNORE NULLS is not supported

I also expect

```sql
❯ select avg(column1 ORDER BY column2) FROM (values (1,2), (3,4), (null,0)) ;
```

to error with "ORDER BY" not supported for avg

### Additional context

@jayzhan211 has some good ideas at https://github.com/apache/arrow-datafusion/pull/9920#discussion_r1549603208 about  how to make checking this easier / harder to miss

**Comment by mustafasrepo** [comment]
I think for `AVG`, or `COUNT` type of aggregates we can safely ignore `ORDER BY` expression. Since operation result is same for any permutation of the input. However, for `first_value`, `nth_value`, `last_value` this is not the case. However, I agree that `IGNORE NULLS` should raise an error.

**Comment by alamb** [comment]
> I think for AVG, or COUNT type of aggregates we can safely ignore ORDER BY expression. Since operation result is same for any permutation of the input.

That is true, though applying an ORDER BY to the argument of COUNT may be much slower 🤔  But maybe we could remove `ORDER BY` that don't change the output as a follow on optimization

**Comment by alamb** [comment]
I wrote some tests for this feature in another PR that we decided not to use. I put them here https://github.com/apache/arrow-datafusion/pull/9953 in case that is helpful @huaxingao

**Comment by mustafasrepo** [comment]
> FWIW @jayzhan211 has a really nice API suggestion here I think https://github.com/apache/arrow-datafusion/pull/9920/files#r1549905825
> 
> Specifically, add these two functions to the `AggregateUDFImpl`
> 
> ```rust
> /// Return true if the aggregate supports `IGNORE NULL`s
> fn support_ignore_nulls() -> bool  { false }
> 
> /// Return true if the aggregate supports and `ORDER BY` to specify its ordering:
> /// `SELECT first_value(x ORDER BY y) ... `
> fn support_ordering() -> bool { false }
> ```

I think these APIs are very nice, clear. We can add these supports

**Comment by vbarua** [comment]
I opened up an issue that's potentially related to this when it comes to supporting IGNORE NULLS
https://github.com/apache/datafusion/issues/15006
